lib/modules/manager/npm/update/dependency/index.ts

```diff
@@ -25,7 +28,16 @@ function replaceAsString(
   }
   // Look for the old version number
   const searchString = `"${oldValue}"`;
-  const newString = `"${newValue}"`;
+  let newString = `"${newValue}"`;
+
+  if (patchReg.test(oldValue)) {
+    const regex = regEx('(patch:' + depName + '@(npm:)?).*#');
+    const patch = oldValue.replace(regex, '$1' + newValue + '#');
```

Comment:
```suggestion
    const regex = regEx(`(?<value>patch:${depName}@(?:npm:)?).*#`);
    const patch = oldValue.replace(regex, `$<value>${newValue}#`);
```

use named capture group, see failed test. We propaböy need to escape `depName` somehow too. 🤔

Reply:
i can be less accurate and go with ".*" intead of "depName", will work too if i need to get rid of the depName

Reply:
Thats why i suggest named capture group, which always work.

Reply:
i fixed it in my own normal way  :D capturing groups dont help me here